src/utils/xmltools.ts

```diff
@@ -264,22 +264,27 @@ function createFieldFromInput(input: Node): GeneratedField {
     maxOccurrences: maximumOccurrencesAttr
       ? Number.parseInt(maximumOccurrencesAttr.value)
       : undefined
-  });
+  },input);
 
   return { name, type, comment, optional };
 }
 
 function getType(
   inputType: string,
-  options: { maxOccurrences?: number }
-): GeneratedFieldType {
-  switch (inputType) {
-    case "ContentSelector":
+  options: { maxOccurrences?: number },
+  input: Node
+): GeneratedFieldType | string  {
+  switch (inputType ? inputType.toLocaleLowerCase() : "") {
+    case "contentselector":
       return options.maxOccurrences === 1
         ? GeneratedFieldType.String
         : GeneratedFieldType.StringArray;
-    case "CheckBox":
+    case "checkbox":
       return GeneratedFieldType.Boolean;
+    case "combobox":
+      return mapXpathResult(
+        evaluate("./config/option", input),
+        (node: Node): Array<string> => {return xpath.select1("@value",node).value}).map(value => `"${value}"`).join(" | ");
```

Comment:
I think giving this a bit more breading room would make it much easier to read.

```typescript
    case "combobox":
      const alternatives = mapXpathResult<Array<string>>(
        evaluate("./config/option", input),
        (node) => xpath.select1("@value", node).value
      );

      return alternatives
        .map(value => `"${value}"`)
        .join(" | ");
```

Reply:
Agree

Reply:
I did the update when I squashed and merged